core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionServer.scala

```diff
@@ -331,3 +345,15 @@ private class ErrorServlet extends RestServlet {
     sendResponse(error, response)
   }
 }
+
+private class ServerStatusServlet(server: RestSubmissionServer) extends RestServlet {
+  override def doGet(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
+    val path = req.getRequestURI
+    if (!server.isServerHealthy() && path == "/health") {
```

Comment:
I would switch the order (check path first). However, from this logic, if server is healthy and request is asking for /health, it will return status instead?